### PR TITLE
Add fix-language extension

### PR DESCRIPTION
**Pull request by hiiamtin**
## Description

**Fix Language** is a Raycast extension that helps you seamlessly switch the language input source of your text when you type and forget to change the language. It supports quick switching for both selected text and clipboard content, making it a useful tool for users who work in multiple languages, such as English and Thai.

## Screencast
image
image2

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/fix-language/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# fix language Changelog
+
+## [Initial Version] - 2567-08-29
```
Comment (pernielsentikaer) [inline extensions/fix-language/CHANGELOG.md:3]:
```suggestion
## [Initial Version] - {PR_MERGE_DATE}
```

File: extensions/fix-language/package.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  "categories": [
+    "Productivity",
+    "Developer Tools",
+    "Other"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "sw",
+      "title": "Sw",
```
Comment (pernielsentikaer) [inline extensions/fix-language/package.json:17]: Could we improve the naming?

File: extensions/fix-language/package.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      "subtitle": "Quick switch the selected text input source",
+      "description": "Switch the language to another (default: eng/thai)",
+      "mode": "no-view"
+    },
+    {
+      "name": "swc",
+      "title": "Swc",
```
Comment (pernielsentikaer) [inline extensions/fix-language/package.json:24]: Could we improve the naming?

File: extensions/fix-language/package.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    {
+      "name": "fix-language",
+      "title": "Fix Language",
+      "subtitle": "Switch the language input source",
+      "description": "Switch the language input source",
+      "mode": "view"
+    },
+    {
+      "name": "fix-language-setting",
+      "title": "Fix Language Setting",
```
Comment (pernielsentikaer) [inline extensions/fix-language/package.json:38]: This should be moved to [preferences](https://developers.raycast.com/api-reference/preferences) instead

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Missing [custom extension](https://icon.ray.so/) icon

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/fix-language/package.json (hunk 2 above)
Comment (hiiamtin) [reply extensions/fix-language/package.json:17]: Sure, I just need something short that I can type quickly when using. Do you have any suggestions for a name or can it be an alias? The name I have in mind is `Switch Selected Language`. and `Switch Clipboard Language` comments below.

File: extensions/fix-language/package.json (hunk 4 above)
Comment (hiiamtin) [reply extensions/fix-language/package.json]: I move it to [preferences](https://developers.raycast.com/api-reference/preferences) instead.

**Comment by hiiamtin** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * Missing [custom extension](https://icon.ray.so/) icon
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

I have already made the correction as you suggested. Please review it again. Thank you.

File: extensions/fix-language/package.json [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      "title": "Fix Language",
+      "subtitle": "Switch the language input source",
+      "description": "Switch the language input source",
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "firstLang",
+      "title": "first language",
```
Comment (pernielsentikaer) [inline extensions/fix-language/package.json:40]:
```suggestion
      "title": "First Language",
```

File: extensions/fix-language/package.json [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        },
+        {
+          "value": "en",
+          "title": "English"
+        }
+      ]
+    },
+    {
+      "name": "secondLang",
+      "title": "second language",
```
Comment (pernielsentikaer) [inline extensions/fix-language/package.json:59]:
```suggestion
      "title": "Second Language",
```

File: extensions/fix-language/src/fix-language.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    } else {
+      setResult("");
+    }
+  };
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Past to Code" onSubmit={handleSubmit} />
```
Comment (pernielsentikaer) [inline extensions/fix-language/src/fix-language.tsx:35]:
```suggestion
          <Action.SubmitForm title="Paste to Active App" onSubmit={handleSubmit} />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Some more feedback 🙂

- Should we consider adding icons to all actions?

**Comment by hiiamtin** [comment]
> Some more feedback 🙂
> 
> * Should we consider adding icons to all actions?

Thank you for the feedback! I plan to add icons along with additional language support in the next release. For this PR, I'd like to focus on the typo fixes to keep the changes minimal and easier to review. Would it be okay to address the icons in a separate PR?

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by pernielsentikaer** [comment]
Just a quick question about the name "Fix Language" might no be saying that much for many, what about some Thai- Eng Text Formatter or something like that?

**Comment by hiiamtin** [comment]
> Just a quick question about the name "Fix Language" might no be saying that much for many, what about some Thai- Eng Text Formatter or something like that?

"Thanks for the suggestion! I’m actually planning to add support for other languages, so I’m concerned that including 'Thai-Eng' in the name might feel limiting in the future. My goal is to create a tool that can switch between multiple language inputs, making it versatile for any language pair. Perhaps a name like 'Language Switcher' or 'Language Fixer' could convey the broader functionality. I'd love to hear your thoughts!"

**Comment by pernielsentikaer** [comment]
Won't it be hard to support many languages?

**Comment by hiiamtin** [comment]
> Won't it be hard to support many languages?

Supporting additional languages isn't too difficult, as I've already tried it briefly, and it seems feasible. However, it will require some time to fully implement. Since I’m currently quite busy, I'll be focusing on the current version for now, but I plan to update and expand the language support in the new version, likely next month(or 2 😅). Thanks for your input!

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**